### ISSUE TITLE
provide docker image

### ISSUE THREAD

**Issue by marcosh** [maglnet/ComposerRequireChecker#78]
It could be really nice if a Docker image was provided, so that one could use a command like

```
docker run -v $(pwd):/app compoesrrequirechecker
```

to run the executable

**Comment by Ocramius** [comment]
Don't think that starting to dockerise CLI projects is a good idea: yes, feasible, but it brings in a lot of additional effort for something that "just works" when having the PHP binary. `docker run -v $(pwd):/app -v path/to/composer-require-checker:/checker php:7.2 /checker/cli/composer-require-checker` works as well

**Comment by marcosh** [comment]
It would be more than enough for me to have the `docker run` command in the readme, so one could use the script without having the need to have the PHP binary installed

**Comment by jaapio** [comment]
someing like `docker run --rm -v $(pwd):/app --workdir=/app php:7.1 php composer-require-checker.phar` ?

**Comment by Ocramius** [comment]
Yeah, but... What's the difference with `cd /your/project && php composer-require-checker.phar` and that? :P

I understand it, but please think about the implication on ecosystem level: does it make sense to have these directions, considering that the entire PHP ecosystem simply assumes that you have a PHP binary of some type (locally installed, remote, VM or docker doesn't matter)?

**Comment by jaapio** [comment]
not everyone is using docker. And I think people who do use docker on a daily base will be able to reproduce the command I added in just a sec.

**Comment by Ocramius** [comment]
> not everyone is using docker.

From my PoV (and I was a nay-sayer before), this is just a question of time.

**Comment by mathroc** [comment]
I don't want to open a new issue for this since it has already been discussed so I'm just posting a comment here in case it could be reopend:

I'd like a docker image because I only want to run this in CI, I can (quite easily) do this multiple ways, eg: by downloading the phar each time in php image. but this come at some cost that is duplicated everywhere this tool is used.

People have already made image for this tool (eg: https://hub.docker.com/r/steevanb/composer-require-checker or https://hub.docker.com/r/webfactory/composer-require-checker/tags) but they are outdated. I or anyone can do it again but it will probably be outdated again later. This makes it *very* easy to integrate in various CI pipelines, the GitLab exemple would look someling like:

```yaml
test:composer-require-check:
  image: maglnet:composer-require-checker
  script: composer-require-checker check composer.json
```

So by having an image, built and distributed automatically by the source project would be a nice improvement.

I can't promise I'll be able to continue maintain it in 10 years but I'm willing to create a merge request and help what I can to integrate this in the project if this become an option

**Comment by Ocramius** [comment]
Maintaining a docker image also means maintaining the whole dependency chain: no thanks.

See also the countless issues around phar builds :-\
